Homework/Nam_Nguyen/Warmup/LeetCode_217_ContainsDuplicate.java

```diff
@@ -0,0 +1,33 @@
+package Homework.Nam_Nguyen.Warmup;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class Solution {
+    public static boolean containsDuplicate(int[] nums) {
+        Map<Integer, Integer> occ = new HashMap<>();
+        int length = nums.length;
+        int mid = length/2;
+        for (int i = 0; i < mid; i++) {
```

Comment:
this algorithm is valid, however, in an interview, it maybe hard to explain to interviewer and easy to create bug as well?
 
I would prefer a single for loop (0 => length-1) for safety. And, the runtime is the same right?

Reply:
Thank you for your comment. I think it's faster to iterator half of the array than the entire array. I tried, and the result is 9ms and 10ms for each of the above solutions. However, in an interview, as you mentioned above, it's better to choose a single for loop in this case.